tests/chainer_tests/iterators_tests/test_multiprocess_iterator.py

```diff
@@ -374,4 +391,162 @@ def test_iterator_serialize_backward_compat(self):
+
+    # unless you're debugging tests, this should be false
+    show_interruption_msg = False
+
+    def setUp(self):
+        self.code_path = None
+        self.sys_path_appended = False
+        if not self.__class__.show_interruption_msg:
+            self.nullfd = os.open(os.devnull, os.O_WRONLY)
+
+    def tearDown(self):
+        if not self.__class__.show_interruption_msg:
+            os.close(self.nullfd)
+        if self.code_path is not None:
+            os.remove(self.code_path)
+        if self.sys_path_appended:
+            sys.path.pop()
+
+    def run_code(self, dataset, n_processes, n_prefetch, shared_mem,
+                 operation):
+        code_template = """
+import random
+import time
+from chainer import iterators
+
+class InfiniteWaitDataSet(object):
+    def __len__(self):
+        return 1000000
+    def __getitem__(self, _):
+        time.sleep(1000000)
+infinite_wait = InfiniteWaitDataSet()
+
+class NoWaitDataSet(object):
+    def __len__(self):
+        return 1000000
+    def __getitem__(self, _):
+        return 0
+no_wait = NoWaitDataSet()
+
+if __name__ == '__main__':
+    if {shared_mem} is not None and {dataset} is infinite_wait:
+        iterators.MultiprocessIterator._interruption_testing = True
+    it = iterators.MultiprocessIterator({dataset}, 100,
+                                        n_processes={n_processes},
+                                        n_prefetch={n_prefetch},
+                                        shared_mem={shared_mem})
+    {operation}
+        """
+        code = code_template.format(dataset=dataset,
+                                    n_processes=n_processes,
+                                    n_prefetch=n_prefetch,
+                                    shared_mem=shared_mem,
+                                    operation=operation)
+        fd, self.code_path = tempfile.mkstemp(suffix='.py')
+        os.write(fd, six.b(code))
+        os.close(fd)
+        sys.path.append(os.path.dirname(self.code_path))
+        self.sys_path_appended = True
+
+        if shared_mem is not None and dataset is 'infinite_wait':
+            stdout = subprocess.PIPE
+        else:
+            stdout = None
+        stderr = None if self.__class__.show_interruption_msg else self.nullfd
+        self.p = subprocess.Popen([sys.executable, self.code_path],
+                                  stdout=stdout, stderr=stderr)
+        if stdout is None:
+            self.child_pids = []
+        else:
+            self.child_pids = list(map(int, self.p.stdout.readline().split()))
+
+    def send_sigint(self):
+        if os.name == 'nt':
+            try:
+                os.kill(self.p.pid, signal.CTRL_C_EVENT)
+                while True:
+                    pass
+            except KeyboardInterrupt:
+                pass
+        else:
+            os.kill(self.p.pid, signal.SIGINT)
+
+    def killall(self):
+        # try waiting the root process
+        for _ in range(10):
```

Comment:
Why do not use `self.p.wait(10)`?

Reply:
Thanks a lot for you suggestion. When I was writing the code, I was preoccupied with the idea of the `wait(1)` → `poll()` pattern to ensure interruptability on any platform.